src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,130 @@
+    void setUp() {
+        user = new User();
+        user.setLogin("Abraham");
+        user.setPassword("fo30re9st");
+        user.setAge(28);
+    }
+
+    @Test
+    void register_NullAge_NotOk() {
+        user.setAge(null);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_NullLogin_NotOK() {
+        user.setLogin(null);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_NullPassword_NotOK() {
+        user.setPassword(null);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_LessLoginLength_NotOk() {
+        user.setLogin("123");
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+        user.setLogin("1234");
+        assertThrows(RegistrationException.class,() -> registrationService.register(user));
+        user.setLogin("12345");
+        assertThrows(RegistrationException.class,() -> registrationService.register(user));
+    }
+
+    @Test
+    void register_LessPasswordLength_NotOk() {
+        user.setPassword("123");
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+        user.setPassword("12345");
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_AgeLessThanRequired_NotOk() {
+        user.setAge(15);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+        user.setAge(7);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+        user.setAge(0);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_AgeGreaterThan18_Ok() {
+        user.setAge(18);
+        assertTrue(registrationService.register(user).getAge()
+                >= RegistrationServiceImpl.MIN_USER_AGE);
+        Storage.people.clear();
+        user.setLogin("Abraham");
+        user.setPassword("fo30re9st");
+        user.setAge(79);
+        assertTrue(registrationService.register(user).getAge()
+                >= RegistrationServiceImpl.MIN_USER_AGE);
+    }
+
+    @Test
+    void register_LoginHasCorrectLength_Ok() {
+        user.setLogin("abcdef");
+        assertTrue(registrationService.register(user).getLogin().length()
+                >= RegistrationServiceImpl.MIN_LOGIN_CHARACTERS);
+        Storage.people.clear();
+        user.setLogin("Bgreekk123565567");
+        user.setPassword("fo30re9st");
+        user.setAge(28);
+        assertTrue(registrationService.register(user).getLogin().length()
+                >= RegistrationServiceImpl.MIN_LOGIN_CHARACTERS);
+    }
+
+    @Test
+    void register_PasswordHasCorrectLength_Ok() {
+        user.setPassword("123456");

```

Comment:
Instead of assertTrue, use assertEquals to directly compare the expected password length with the actual password length returned by the register method.